### PR TITLE
chore(infra): Types for ingress and more

### PR DESCRIPTION
**Pull request by AndesKrrrrrrrrrrr**
Add some typing so that our IDEs warn when using invalid values

## Summary by CodeRabbit

- **New Features**
	- Introduced new ingress configurations, categorizing them as 'internal' for better clarity.
	- Added a new type `IngressType` for improved semantic alignment in ingress configurations.

- **Improvements**
	- Enhanced clarity in type definitions related to environment configurations, particularly for PostgreSQL and Redis information.
	- Updated naming conventions in type declarations for consistency and clarity.

- **Bug Fixes**
	- Resolved inconsistencies in naming conventions across various types and interfaces.

### CODE REVIEW

File: apps/web/infra/web.ts [hunk 1]
```diff
@@ -49,7 +50,7 @@ export const serviceSetup = (services: {
           },
         },
         paths: ['/'],
-      },
+      } as Ingress,
```
Comment (busla) [inline apps/web/infra/web.ts:53]: consider adjusting the implementation to prevent typecasting

File: infra/src/dsl/dsl.ts [hunk 2]
```diff
@@ -65,7 +66,7 @@ export class ServiceBuilder<ServiceType extends string> {
       grantNamespaces: [],
       grantNamespacesEnabled: false,
       secrets: COMMON_SECRETS,
-      ingress: {},
+      ingress: {} as IngressMapping, // Force type conformance for initialization
```
Comment (busla) [inline infra/src/dsl/dsl.ts:69]: another type casting here

**Review by busla — CHANGES_REQUESTED** [review]
Not sure what benefits this PR is introducing.

**Comment by AndesKrrrrrrrrrrr** [comment]
> Not sure what benefits this PR is introducing.

Cf. the PR description:
> Add some typing so that our IDEs warn when using invalid values

Without this typing one could trivially add invalid ingress config, without realizing it. https://github.com/island-is/island.is/pull/16404, for example used an invalid ingress, but without this typing, none would realize.

**Review by busla — APPROVED** [review]
LGTM

**Pull request merged**